### ISSUE TITLE
Bot randomly stops matchmaking and accepting requests

### ISSUE THREAD

**Issue by Quintium** [ShailChoksi/lichess-bot#607]
A day or two after starting, the bot hangs and stops matchmaking and accepting requests. This was already reported in #573 , but the author marked it as closed after an update had apparently fixed the bug. The bug stills happens to me even after re-cloning the repo several times.

Here's the verbose logs:
```                      
[11/30/22 22:35:06] DEBUG    <UciProtocol (pid=15357)>: >> info      engine.py:1035
                             score cp -1288 depth 1 nodes 71 time 33               
                             nps 2113 pv                                           
                    DEBUG    <UciProtocol (pid=15357)>: >> info      engine.py:1035
                             score cp -1288 depth 2 nodes 203 time                 
                             34 nps 5867 pv                                        
                    DEBUG    <UciProtocol (pid=15357)>: >> info      engine.py:1035
                             score cp -1399 depth 3 nodes 1208 time                
                             40 nps 29499 pv                                       
                    DEBUG    <UciProtocol (pid=15357)>: >> info      engine.py:1035
                             score cp -1807 depth 4 nodes 7769 time                
                             61 nps 126721 pv                                      
                    DEBUG    <UciProtocol (pid=15357)>: >> info      engine.py:1035
                             score cp -1932 depth 5 nodes 12905 time               
                             88 nps 145648 pv                                      
                    DEBUG    <UciProtocol (pid=15357)>: >> info      engine.py:1035
                             score cp -2458 depth 6 nodes 26965 time               
                             128 nps 210047 pv                                     
                    DEBUG    <UciProtocol (pid=15357)>: >> info      engine.py:1035
                             score mate -3 depth 7 nodes 42325 time                
                             200 nps 211608 pv d1d7 a4d7 g7g6 g5g6                 
                             g8h8 g6g7                                             
                    DEBUG    <UciProtocol (pid=15357)>: >> bestmove  engine.py:1035
                             d1d7                                                  
                    INFO     Evaluation: #-3                  engine_wrapper.py:264
                    INFO     Depth: 7                         engine_wrapper.py:264
                    INFO     Nodes: 42.3K                     engine_wrapper.py:264
                    INFO     Speed: 211.6Knps                 engine_wrapper.py:264
                    INFO     PV: 69...Rd7+ 70. Bxd7 g6 71.    engine_wrapper.py:264
                             Qxg6+ Kh8 72. Qg7#                                    
                    DEBUG    https://lichess.org:443 "POST    connectionpool.py:456
                             /api/bot/game/CHdWV3Jf/move/d1d7                      
                             ?offeringDraw=false HTTP/1.1"                         
                             200 11                                                
                    DEBUG    Game state: {'type': 'gameState',   lichess-bot.py:536
                             'moves': 'e2e4 c7c5 g1f3 d7d6 d2d4                    
                             c5d4 f3d4 g8f6 b1c3 a7a6 f1e2 e7e5                    
                             d4b3 b8c6 g2g4 f8e7 g4g5 f6d7 c3d5                    
                             e8g8 d5e7 d8e7 c1e3 d7c5 b3c5 d6c5                    
                             c2c4 f8d8 d1c1 c6d4 e2f1 d4f3 e1e2                    
                             f3d4 e3d4 d8d4 e2f3 c8e6 h1g1 a8d8                    
                             c1c3 e7e8 a1e1 e8a4 e1a1 d8e8 b2b3                    
                             a4b4 c3b4 c5b4 f3e3 e8d8 f1e2 e6d7                    
                             g1d1 d7c6 d1d4 e5d4 e3d3 d8d6 a1e1                    
                             d6g6 h2h4 h7h6 d3d4 h6g5 h4g5 g6g5                    
                             e2f3 g5a5 e1e2 f7f5 f3h1 g8h7 f2f3                    
                             f5f4 c4c5 h7g8 d4c4 a5b5 h1g2 c6d7                    
                             c4d5 d7e8 d5d6 b5a5 e4e5 e8b5 e2d2                    
                             b5e8 e5e6 e8c6 e6e7 c6e8 d2e2 g8f7                    
                             g2h3 f7g6 h3e6 g6h7 e6d7 e8f7 e7e8n                   
                             f7g6 e8c7 g6b1 d7a4 b1d3 e2e7 b7b5                    
                             c5b6 a5g5 c7e8 h7g8 e8c7 g5g6 c7e6                    
                             d3f5 b6b7 g8h7 b7b8q f5e6 e7e6 g6g2                   
                             b8b4 g2a2 b4f4 a2b2 f4g5 b2h2 f3f4                    
                             h2h1 f4f5 h7g8 f5f6 h1d1 d6e7                         
                             d1d7', 'wtime': 170, 'btime':                         
                             28680, 'winc': 0, 'binc': 0,                          
                             'status': 'started'}                                  
                    DEBUG    Game state: {'type': 'gameState',   lichess-bot.py:536
                             'moves': 'e2e4 c7c5 g1f3 d7d6 d2d4                    
                             c5d4 f3d4 g8f6 b1c3 a7a6 f1e2 e7e5                    
                             d4b3 b8c6 g2g4 f8e7 g4g5 f6d7 c3d5                    
                             e8g8 d5e7 d8e7 c1e3 d7c5 b3c5 d6c5                    
                             c2c4 f8d8 d1c1 c6d4 e2f1 d4f3 e1e2                    
                             f3d4 e3d4 d8d4 e2f3 c8e6 h1g1 a8d8                    
                             c1c3 e7e8 a1e1 e8a4 e1a1 d8e8 b2b3                    
                             a4b4 c3b4 c5b4 f3e3 e8d8 f1e2 e6d7                    
                             g1d1 d7c6 d1d4 e5d4 e3d3 d8d6 a1e1                    
                             d6g6 h2h4 h7h6 d3d4 h6g5 h4g5 g6g5                    
                             e2f3 g5a5 e1e2 f7f5 f3h1 g8h7 f2f3                    
                             f5f4 c4c5 h7g8 d4c4 a5b5 h1g2 c6d7                    
                             c4d5 d7e8 d5d6 b5a5 e4e5 e8b5 e2d2                    
                             b5e8 e5e6 e8c6 e6e7 c6e8 d2e2 g8f7                    
                             g2h3 f7g6 h3e6 g6h7 e6d7 e8f7 e7e8n                   
                             f7g6 e8c7 g6b1 d7a4 b1d3 e2e7 b7b5                    
                             c5b6 a5g5 c7e8 h7g8 e8c7 g5g6 c7e6                    
                             d3f5 b6b7 g8h7 b7b8q f5e6 e7e6 g6g2                   
                             b8b4 g2a2 b4f4 a2b2 f4g5 b2h2 f3f4                    
                             h2h1 f4f5 h7g8 f5f6 h1d1 d6e7                         
                             d1d7', 'wtime': 0, 'btime': 28680,                    
                             'winc': 0, 'binc': 0, 'status':                       
                             'outoftime', 'winner': 'black'}                       
                    DEBUG    <UciProtocol (pid=15357)>: << position  engine.py:1009
                             startpos moves e2e4 c7c5 g1f3 d7d6 d2d4               
                             c5d4 f3d4 g8f6 b1c3 a7a6 f1e2 e7e5 d4b3               
                             b8c6 g2g4 f8e7 g4g5 f6d7 c3d5 e8g8 d5e7               
                             d8e7 c1e3 d7c5 b3c5 d6c5 c2c4 f8d8 d1c1               
                             c6d4 e2f1 d4f3 e1e2 f3d4 e3d4 d8d4 e2f3               
                             c8e6 h1g1 a8d8 c1c3 e7e8 a1e1 e8a4 e1a1               
                             d8e8 b2b3 a4b4 c3b4 c5b4 f3e3 e8d8 f1e2               
                             e6d7 g1d1 d7c6 d1d4 e5d4 e3d3 d8d6 a1e1               
                             d6g6 h2h4 h7h6 d3d4 h6g5 h4g5 g6g5 e2f3               
                             g5a5 e1e2 f7f5 f3h1 g8h7 f2f3 f5f4 c4c5               
                             h7g8 d4c4 a5b5 h1g2 c6d7 c4d5 d7e8 d5d6               
                             b5a5 e4e5 e8b5 e2d2 b5e8 e5e6 e8c6 e6e7               
                             c6e8 d2e2 g8f7 g2h3 f7g6 h3e6 g6h7 e6d7               
                             e8f7 e7e8n f7g6 e8c7 g6b1 d7a4 b1d3                   
                             e2e7 b7b5 c5b6 a5g5 c7e8 h7g8 e8c7 g5g6               
                             c7e6 d3f5 b6b7 g8h7 b7b8q f5e6 e7e6                   
                             g6g2 b8b4 g2a2 b4f4 a2b2 f4g5 b2h2 f3f4               
                             h2h1 f4f5 h7g8 f5f6 h1d1 d6e7 d1d7                    
                    INFO     QuintBot won!                       lichess-bot.py:609
                    INFO     notropis forfeited on time.         lichess-bot.py:621
                    INFO     ***                                 conversation.py:46
                             https://lichess.org/CHdWV3Jf/black                    
                             [player] QuintBot: Good game!                         
                    DEBUG    https://lichess.org:443 "POST    connectionpool.py:456
                             /api/bot/game/CHdWV3Jf/chat                           
                             HTTP/1.1" 200 11                                      
                    INFO     ***                                 conversation.py:46
                             https://lichess.org/CHdWV3Jf/black                    
                             [spectator] QuintBot: Thanks for                      
                             watching!                                             
                    DEBUG    https://lichess.org:443 "POST    connectionpool.py:456
                             /api/bot/game/CHdWV3Jf/chat                           
                             HTTP/1.1" 200 11                                      
[11/30/22 22:35:07] DEBUG    https://lichess.org:443 "GET     connectionpool.py:456
                             /api/account/playing HTTP/1.1"                        
                             200 17                                                
                    DEBUG    <UciProtocol (pid=15357)>: << stop      engine.py:1009
                    DEBUG    <UciProtocol (pid=15357)>: << quit      engine.py:1009
                    DEBUG    <UciProtocol (pid=15357)>: Process      engine.py:1006
                             exited                                                
                    DEBUG    <UciProtocol (pid=15357)>: Connection    engine.py:993
                             lost (exit code: 1, error: None)                      
                    INFO     ---                                 lichess-bot.py:589
                             https://lichess.org/CHdWV3Jf/black                    
                             Game over                                             
[12/01/22 07:56:41] DEBUG    Starting new HTTPS connection   connectionpool.py:1003
                             (1): lichess.org:443                                  
[12/01/22 07:56:42] DEBUG    https://lichess.org:443 "GET     connectionpool.py:456
                             /api/stream/event HTTP/1.1" 502                       
                             1644                                                  
                    DEBUG    Starting new HTTPS connection   connectionpool.py:1003
                             (1): lichess.org:443                                  
[12/01/22 07:56:43] DEBUG    https://lichess.org:443 "GET     connectionpool.py:456
                             /api/stream/event HTTP/1.1" 502                       
                             1644                                                  
                    DEBUG    Starting new HTTPS connection   connectionpool.py:1003
                             (1): lichess.org:443                                  
[12/01/22 07:56:45] DEBUG    https://lichess.org:443 "GET     connectionpool.py:456
                             /api/stream/event HTTP/1.1" 502                       
                             1644                                                  
                    DEBUG    Starting new HTTPS connection   connectionpool.py:1003
                             (1): lichess.org:443                                  
[12/01/22 07:56:46] DEBUG    https://lichess.org:443 "GET     connectionpool.py:456
                             /api/stream/event HTTP/1.1" 502                       
                             1644                                                  
                    DEBUG    Starting new HTTPS connection   connectionpool.py:1003
                             (1): lichess.org:443                                  
[12/01/22 07:56:47] DEBUG    https://lichess.org:443 "GET     connectionpool.py:456
                             /api/stream/event HTTP/1.1" 502                       
                             1644                                                  
                    DEBUG    Starting new HTTPS connection   connectionpool.py:1003
                             (1): lichess.org:443                                  
[12/01/22 07:56:48] DEBUG    https://lichess.org:443 "GET     connectionpool.py:456
                             /api/stream/event HTTP/1.1" 502                       
                             1644                                                  
                    DEBUG    Starting new HTTPS connection   connectionpool.py:1003
                             (1): lichess.org:443                                  
[12/01/22 07:56:49] DEBUG    https://lichess.org:443 "GET     connectionpool.py:456
                             /api/stream/event HTTP/1.1" 502                       
                             1644                                                  
                    DEBUG    Starting new HTTPS connection   connectionpool.py:1003
                             (1): lichess.org:443                                  
[12/01/22 07:56:50] DEBUG    https://lichess.org:443 "GET     connectionpool.py:456
                             /api/stream/event HTTP/1.1" 502                       
                             1644                                                  
                    DEBUG    Starting new HTTPS connection   connectionpool.py:1003
                             (1): lichess.org:443                                  
[12/01/22 07:56:51] DEBUG    https://lichess.org:443 "GET     connectionpool.py:456
                             /api/stream/event HTTP/1.1" 502                       
                             1644                                                  
                    DEBUG    Starting new HTTPS connection   connectionpool.py:1003
                             (1): lichess.org:443                                  
[12/01/22 07:56:52] DEBUG    https://lichess.org:443 "GET     connectionpool.py:456
                             /api/stream/event HTTP/1.1" 502                       
                             1644                                                  
                    DEBUG    Starting new HTTPS connection   connectionpool.py:1003
                             (1): lichess.org:443                                  
                    DEBUG    https://lichess.org:443 "GET     connectionpool.py:456
                             /api/stream/event HTTP/1.1" 502                       
                             1644                                                  
                    DEBUG    Starting new HTTPS connection   connectionpool.py:1003
                             (1): lichess.org:443                                  
[12/01/22 07:56:53] DEBUG    https://lichess.org:443 "GET     connectionpool.py:456
                             /api/stream/event HTTP/1.1" 502                       
                             1644                                                  
                    DEBUG    Starting new HTTPS connection   connectionpool.py:1003
                             (1): lichess.org:443                                  
[12/01/22 07:56:54] DEBUG    https://lichess.org:443 "GET     connectionpool.py:456
                             /api/stream/event HTTP/1.1" 502                       
                             1644                                                  
                    DEBUG    Starting new HTTPS connection   connectionpool.py:1003
                             (1): lichess.org:443                                  
[12/01/22 07:57:00] DEBUG    https://lichess.org:443 "GET     connectionpool.py:456
                             /api/stream/event HTTP/1.1" 200                       
                             None
```

**Comment by MarkZH** [comment]
I think the latest update should address this problem--specifically PR #612. Please report if lichess-bot is still hanging after updating.

**Comment by Quintium** [comment]
Seems to have been fixed by #612, my bot has now been running for a few days without problems. I will reopen if this happens again.
